overlord/configstate/config/helpers.go

```diff
@@ -157,11 +157,19 @@ func SetSnapConfig(st *state.State, snapName string, snapcfg json.RawMessage) er
 	var config map[string]*json.RawMessage
 	err := st.Get("config", &config)
 	if err == state.ErrNoState {
+		if len(snapcfg) == 0 {
+			// bail out early
```

Comment:
Hmm, will this DTRT if snap cfg is empty and we are setting something? Why is this changed?